### PR TITLE
RDBC-171: unify execute_* methods

### PR DESCRIPTION
**Pull request by dabroz**

### CODE REVIEW

**Comment by dabroz** [comment]
There is a small issue with HTTPS. I'll fix that tomorrow.

**Comment by dabroz** [comment]
So this is weird, `firstTopologyUpdate` consumes the `AuthorizationException` and supresses it in 

```
   if (initialUrls != null && initialUrls.length > 0) {
                initializeUpdateTopologyTimer();
                return;
            }
```

I will verify the test itself with the Java client later, for now I've made a simple fix.

File: lib/requests/request_executor.rb [hunk 1]
```diff
@@ -140,6 +138,8 @@ def new_first_topology_update(input_urls:)
             @_topology_taken_from_node = server_node
 
             true
+          rescue AuthorizationException
```
Comment (gregolsky) [inline lib/requests/request_executor.rb:141]: Is there a difference in beviour here if you did not catch it?

**Review by gregolsky — APPROVED** [review]

File: lib/requests/request_executor.rb (hunk 1 above)
Comment (dabroz) [reply lib/requests/request_executor.rb:141]: Yes, `firstTopologyUpdate` would consume it and the app/test would continue, breaking on the first "real" command. There is something missing from Java client to handle it here and I will investigate it in the future.

Comment (gregolsky) [reply lib/requests/request_executor.rb:141]: It's handled in C#, but not in Java. I'll make a ticket. Take a look:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Http/RequestExecutor.cs#L480

Note `_lastKnownUrls` are updated in those cases.

**Pull request merged**